agent/internal/docker_credential_store.go

```diff
@@ -0,0 +1,82 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	hclient "github.com/docker/docker-credential-helpers/client"
+	"github.com/docker/docker/api/types"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+const (
+	dockerConfigFile        = "/root/.docker/config.json"
+	credentialsHelperPrefix = "docker-credential-"
+	tokenUsername           = "<token>"
+)
+
+type credentialStore struct {
+	registry string
+	store    hclient.ProgramFunc
+}
+
+// newCredentialStore returns the credential helper configured for the
+// given registry in the default docker config or an error.
+func newCredentialStore(ctx *actor.Context, registryHostname string) (*credentialStore, bool) {
+	type ConfigFile struct {
+		CredentialHelpers map[string]string `json:"credHelpers,omitempty"`
+	}
+
+	configFile, err := os.Open(dockerConfigFile)
+	if err != nil {
+		ctx.Log().Warn(err, "can't open docker config")
```

Comment:
You should bubble up all these warnings and wrap them in the caller.

Reply:
It seems like a failure mode if we can't initialize the store (instead of just a warning)

Reply:
This issue is that I don't think the docker config will exist in all cases. If they haven't installed the cli and run configure-docker from a credentials helper (what they would currently need to do), there won't be a config or helpers. So I thought a warn or info and just running with none makes sense.